src/input/error.rs

```diff
@@ -27,6 +29,9 @@ impl Display for AudioStreamError {
             Self::RetryIn(t) => f.write_fmt(format_args!("retry in {:.2}s", t.as_secs_f32())),
             Self::Fail(why) => f.write_fmt(format_args!("{why}")),
             Self::Unsupported => f.write_str("operation was not supported"),
+            Self::HttpRequestFailed(code) => f.write_fmt(format_args!(
+                "Got HTTP Response Code: {code}. You may be getting rate limited."
```

Comment:
In case you want to modify this setup into a file-local error (e.g., an `Error` type in `http.rs` that isn't exported), the convention is that `Display` strings for error types are written in lower case. See [the trait documentation](https://doc.rust-lang.org/std/error/trait.Error.html) -- "Error messages are typically concise lowercase sentences without trailing punctuation".